wsmaster/che-core-api-auth-gitlab/src/main/java/org/eclipse/che/security/oauth/GitLabOAuthAuthenticator.java

```diff
@@ -97,6 +102,28 @@ public OAuthToken getToken(String userId) throws IOException {
     return token;
   }
 
+  @Override
+  public boolean invalidateToken(String token) {
+    HttpClient client = HttpClient.newHttpClient();
+    HttpRequest request =
+        HttpRequest.newBuilder()
+            .uri(
+                URI.create(
+                    format(
+                        "%s/oauth/revoke?client_id=%s&client_secret=%s&token=%s",
+                        gitlabEndpoint, clientId, clientSecret, token)))
+            .POST(HttpRequest.BodyPublishers.noBody())
+            .build();
+
+    try {
+      HttpResponse<InputStream> response =
+          client.send(request, HttpResponse.BodyHandlers.ofInputStream());
+      return response.statusCode() == 200;
```

Comment:
What does it return if token not found (already removed) ?

Reply:
We can not remove an oauth token from Gitlab. If we manually revoke the authorisation from Gitlab, and then execute the revoke action from che dashboard, we will have status code 200.